CMakeLists.txt

```diff
@@ -49,13 +55,44 @@ execute_process(
 )
 find_package(nanobind CONFIG REQUIRED)
 
+if(BUILD_WITH_CHOLMOD_SUPPORT)
+  set(
+    CMAKE_MODULE_PATH
+    ${JRL_CMAKE_MODULES}/find-external/CHOLMOD
+    ${CMAKE_MODULE_PATH}
+  )
+  ADD_PROJECT_DEPENDENCY(CHOLMOD REQUIRED FIND_EXTERNAL "CHOLMOD")
+  message(
+    STATUS
+    "Build with CHOLMOD support (LGPL). See CHOLMOD/Doc/License.txt for further details."
+  )
+  add_definitions(-DNANOEIGENPY_WITH_CHOLMOD_SUPPORT)
+endif(BUILD_WITH_CHOLMOD_SUPPORT)
+
 # Setup main targets
 file(
   GLOB_RECURSE ${PROJECT_NAME}_HEADERS
   CONFIGURE_DEPENDS
   include/nanoeigenpy/*.hpp
 )
 
+list(
+  FILTER ${PROJECT_NAME}_HEADERS
+  EXCLUDE
+  REGEX "include/nanoeigenpy/decompositions/sparse/cholmod/.*"
+)
+if(BUILD_WITH_CHOLMOD_SUPPORT)
+  file(
+    GLOB ${PROJECT_NAME}_DECOMPOSITIONS_SPARSE_CHOLMOD_HEADERS
+    include/nanoeigenpy/decompositions/sparse/cholmod/*.hpp
+  )
+  list(
+    APPEND
+    ${PROJECT_NAME}_HEADERS
+    ${${PROJECT_NAME}_DECOMPOSITIONS_SPARSE_CHOLMOD_HEADERS}
+  )
```

Comment:
Easier to just perform the filter operation in the `else()` branch of the if statement here.